src/Type/Infer.hs

```diff
@@ -540,7 +541,7 @@ inferIsolated contextRange range body inf
          Nothing   -> return res
          Just vrng -> do seff <- subst ieff
                          let (ls,tl) = extractOrderedEffect seff
-                         case filter (\l -> labelName l == nameTpLocal) ls of
+                         case filter (\l -> labelName l == nameTpLocal) (map snd ls) of
```

Comment:
All other changes just look like this essentially (getting the result of `extractOrderedEffect` as it was previously.